src/hyperlight_guest_bin/build.rs

```diff
@@ -78,37 +76,26 @@ fn cargo_main() {
             .flag("-Wno-unused-parameter")
             .flag("-Wno-string-plus-int");
 
-        if is_pe {
-            cfg.flag("-Wno-unused-label");
-            cfg.flag("-Wno-unused-variable");
-            cfg.compiler("clang-cl");
-        } else {
-            cfg.flag("-fPIC");
-            // This is a terrible hack, because
-            // - we need stack clash protection, because we have put the
-            //   stack right smack in the middle of everything in the guest
-            // - clang refuses to do stack clash protection unless it is
-            //   required by a target ABI (Windows, MacOS) or the target is
-            //   is Linux or FreeBSD (see Clang.cpp RenderSCPOptions
-            //   https://github.com/llvm/llvm-project/blob/1bb52e9/clang/lib/Driver/ToolChains/Clang.cpp#L3724).
-            //   Hopefully a flag to force stack clash protection on generic
-            //   targets will eventually show up.
-            cfg.flag("--target=x86_64-unknown-linux-none");
-
-            // We don't support stack protectors at the moment, but Arch Linux clang
-            // auto-enables them for -linux platforms, so explicitly disable them.
-            cfg.flag("-fno-stack-protector");
-            cfg.flag("-fstack-clash-protection");
-            cfg.flag("-mstack-probe-size=4096");
-            cfg.compiler("clang");
-        }
-
-        if cfg!(windows) {
-            unsafe { env::set_var("AR_x86_64_unknown_none", "llvm-ar") };
-        } else {
-            unsafe { env::set_var("AR_x86_64_pc_windows_msvc", "llvm-lib") };
-        }
-
+        cfg.flag("-fPIC");
+        // This is a terrible hack, because
+        // - we need stack clash protection, because we have put the
+        //   stack right smack in the middle of everything in the guest
+        // - clang refuses to do stack clash protection unless it is
+        //   required by a target ABI (Windows, MacOS) or the target is
+        //   is Linux or FreeBSD (see Clang.cpp RenderSCPOptions
+        //   https://github.com/llvm/llvm-project/blob/1bb52e9/clang/lib/Driver/ToolChains/Clang.cpp#L3724).
+        //   Hopefully a flag to force stack clash protection on generic
+        //   targets will eventually show up.
+        cfg.flag("--target=x86_64-unknown-linux-none");
+
+        // We don't support stack protectors at the moment, but Arch Linux clang
+        // auto-enables them for -linux platforms, so explicitly disable them.
+        cfg.flag("-fno-stack-protector");
+        cfg.flag("-fstack-clash-protection");
+        cfg.flag("-mstack-probe-size=4096");
+        cfg.compiler("clang");
+
+        unsafe { env::set_var("AR_x86_64_unknown_none", "llvm-ar") };
```

Comment:
is this correct? I think we might want to keep the if/else block here

Reply:
fixed